### ISSUE TITLE
Wrong check result

### ISSUE THREAD

**Issue by iranimij** [yireo/Yireo_ExtensionChecker#44]
I have two modules, one in `app/code` and another one in my custom route `src` that is defined correctly in the main composer.
I have created a module in `src` folder and then required a constant from the `src` folder into the module that is in `app/code` folder.

so theoretically the module that is in app/code should have a dependency

but in practice Yireo extension checker does not find it and I think it's a bug.

**Comment by jissereitsma** [comment]
I think it is a bug too. Could you share the PHP code with which you have linked module A to module B? If you are worried about namespaces uncovering specific details, just rewrite them to dummy namespaces.

**Comment by iranimij** [comment]
Sure, 

```

<?php

namespace Mainwebsite\AsyncTasks\Model;

use Magento\Framework\App\Config\ScopeConfigInterface;
use Magento\Framework\Exception\NoSuchEntityException;
use Magento\Sales\Api\OrderRepositoryInterface;
use Mainwebsite\Configuration\Model\GeneralConfig;

class Consumer
{
    /**
     * @var OrderRepositoryInterface
     */
    protected $orderRepository;

    /**
     * @var \Klaviyo\Reclaim\Helper\Data
     */
    protected $klaviyoHelper;

    protected $scopeConfig;

    /**
     * Consumer constructor.
     *
     * @param OrderRepositoryInterface $orderRepository
     * @param \Klaviyo\Reclaim\Helper\Data $klaviyoHelper
     */
    public function __construct(
        OrderRepositoryInterface $orderRepository,
        \Klaviyo\Reclaim\Helper\Data $klaviyoHelper,
        ScopeConfigInterface $scopeConfig
    ) {
        $this->orderRepository = $orderRepository;
        $this->klaviyoHelper = $klaviyoHelper;
        $this->scopeConfig = $scopeConfig;
    }

    /**
     * @param string $orderId
     */
    public function processOrder(string $orderId)
    {
        $order = $this->orderRepository->get((int)$orderId);
        $this->addCustomerToKlaviyoPurchaserList($order);
        $this->sendSlackNotificationAboutOrder($order);
        $this->reduceRentalQtyForProducts($order);
    }

    protected function reduceRentalQtyForProducts($order)
    {
        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();

        foreach ($order->getAllItems() as $orderItem) {
            $sku = $orderItem->getData('sku');
            if (stristr($sku, '-mieten') !== false && stristr($sku, '-neuware') === false) {
                // Nicht für Neuware-mieten
                $product = $objectManager->get('\Magento\Catalog\Api\ProductRepositoryInterface');
                try {
                    $product = $product->get($orderItem->getProduct()->getSku());
                } catch (NoSuchEntityException $e) {
                }

                if ($product->getId()) {
                    #var_dump($product->getSku(), ($product->getQtyRental() - $orderItem->getQtyOrdered()));
                    $productAction = $objectManager->get('\Magento\Catalog\Model\ResourceModel\Product\Action');
                    $productAction->updateAttributes(
                        [$product->getId()],
                        [
                            'qty_rental' => max(0, ($product->getQtyRental() - $orderItem->getQtyOrdered()))
                        ],
                        \Magento\Store\Model\Store::DEFAULT_STORE_ID
                    );
                    $flushCacheProduct = $product;

                    // Flush cache
                    $cacheManager = $objectManager->get('\Magento\Framework\App\CacheInterface');
                    $cacheManager->clean('catalog_product_' . $flushCacheProduct->getId());
                    $cacheManager->clean('cat_p_' . $flushCacheProduct->getId());
                    $flushCacheProduct->cleanCache();
                    $observer = new \Magento\Framework\Event\Observer(
                        [
                            'event' => new \Magento\Framework\DataObject(
                                [
                                    'object' => $flushCacheProduct
                                ]
                            )
                        ]
                    );
                    $flushCacheObserver = $objectManager->create('\Magento\PageCache\Observer\FlushCacheByTags');
                    $flushCacheObserver->execute($observer);
                    $cacheContext = $objectManager->get('Magento\Framework\Indexer\CacheContext');
                    $cacheContext->registerEntities(\Magento\Catalog\Model\Product::CACHE_TAG, array_unique([$flushCacheProduct->getId()]));
                    $objectManager->get('Magento\Framework\Event\ManagerInterface')->dispatch('clean_cache_by_tags', ['object' => $cacheContext]);
                    $cacheManager->clean('mainwebsite_configurator');
                }
            }
        }
    }

    protected function addCustomerToKlaviyoPurchaserList($order)
    {
        $this->klaviyoHelper->subscribeEmailToKlaviyoPurchasersList($order->getCustomerEmail(), $order->getCustomerFirstname(), $order->getCustomerLastname());
    }

    /**
     * @param \Magento\Sales\Model\Order $order
     */
    protected function sendSlackNotificationAboutOrder($order)
    {
        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
        $shipToCountry = $order->getBillingAddress()->getCountryId();
        if ($order->getShippingAddress()) {
            $shipToCountry = $order->getShippingAddress()->getCountryId();
        }
        $deliveryInfo = $order->getWuwiDeliveryInformation();
        if (!empty($deliveryInfo)) {
            $deliveryInfo = str_replace(['Lieferung in ', 'Werktagen'], ['', 'Werktage'], $deliveryInfo);
            // Count orders by customer
            $nthOrder = $objectManager->get('\Magento\Sales\Model\OrderFactory')->create()->getCollection()->addAttributeToFilter('customer_email', $order->getCustomerEmail())->count();
            $deliveryInfo = ' (#'.$nthOrder.', '.$deliveryInfo.')';
        }
        $msgText = $shipToCountry . ': ' . $order->getCustomerFirstname() . ' ' . $order->getCustomerLastname() . $deliveryInfo;
        $productsFound = false;
        foreach ($order->getAllItems() as $item) {
            $product = $item->getProduct();
            if ($product->getTypeId() == 'configurable') continue;
            if (stristr($item->getSku(), 'label-') !== false) continue;
            if (!empty($msgText)) {
                $msgText .= "\n";
            }
            $productsFound = true;
            $price = $item->getBaseRowTotalInclTax();
            if (!$price && $item->getParentItem()) {
                $price = $item->getParentItem()->getBaseRowTotalInclTax();
            }
            if ($item->getParentItem()) {
                $isBrandNewRental = stristr($item->getParentItem()->getSku(), '-neuware') !== false;
                $isRental         = stristr($item->getParentItem()->getSku(), '-mieten') !== false;
            } else {
                $isBrandNewRental = stristr($item->getSku(), '-neuware') !== false;
                $isRental         = stristr($item->getSku(), '-mieten') !== false;
            }
            $conditionString = $isRental ? 'M' : 'K';
            if ($isRental) {
                $conditionString .= $isBrandNewRental ? '-NW' : '-PL';
            }
            $msgText .= round($item->getQtyOrdered()) . 'x ' . $item->getName() . ' ('.$conditionString.', ' . strip_tags($order->formatBasePrice($price)) . ')';
        }
        if (!$productsFound) {
            // Dont send notification for DHL labels, etc.
            return $this;
        }
        $couponCode = '';
        if ($order->getCouponCode()) {
            $couponCode = preg_replace('/[^0-9A-Za-z-\.\&\+\*_\ö\ä\ü\Ö\Ä\Ü\ß ]/', '', $order->getCouponCode());
            $resource = $objectManager->get('Magento\Framework\App\ResourceConnection');
            $connection = $resource->getConnection();
            $tblSalesOrder = $connection->getTableName('sales_order');
            $usageCount = $connection->fetchOne('select count(*) from `'.$tblSalesOrder.'` where coupon_code = "'.$couponCode.'" group by coupon_code') ?: 0;
            $couponCode = ' ('.$order->getCouponCode().' ('.$usageCount.'x))';
        }
        $color = '#00ff00'; // Green
        $slackUrl = $this->scopeConfig->getValue(GeneralConfig::SECOND_SLACK_URL);
        $messageArray = [
            //'channel' => "@sebastian",
            'link_names' => 1,
            'username' => "Mainwebsite-Shop",
            "icon_emoji" => ':tada:',
            'attachments' => [
                [
                    'title' => 'Bestellung #' . $order->getIncrementId() . ': ' . strip_tags($order->formatBasePrice($order->getBaseGrandTotal())) . $couponCode,
                    'color' => $color,
                    'text' => $msgText,
                    "mrkdwn_in" => [
                        "text",
                        "pretext"
                    ]
                ]
            ]
        ];
        $this->sendToSlack($slackUrl, $messageArray);
    }

    protected function sendToSlack($webhookUrl, $data)
    {
        $postData = array(
            'payload' => json_encode($data),
        );

        $ch = curl_init();

        curl_setopt($ch, CURLOPT_URL, $webhookUrl);
        curl_setopt($ch, CURLOPT_POST, 1);
        curl_setopt($ch, CURLOPT_TIMEOUT, 20);
        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
        curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($postData));

        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

        $server_output = curl_exec($ch);

        curl_close($ch);

        if (trim($server_output) == 'ok') {
            return true;
        }

        return $server_output;
    }
}
`

**Comment by iranimij** [comment]
Below code is from a module inside `app/code` and it uses `Wunderwiege\Configuration\Model\GeneralConfig` that is in src folder.
```
<?php

namespace Wunderwiege\AsyncTasks\Model;

use Magento\Framework\App\Config\ScopeConfigInterface;
use Magento\Framework\Exception\NoSuchEntityException;
use Magento\Sales\Api\OrderRepositoryInterface;
use Wunderwiege\Configuration\Model\GeneralConfig;

echo GeneralConfig::SECOND_SLACK_URL;
```

May I work on this issue and send my PR ?

**Comment by jissereitsma** [comment]
Thanks. Yes, this looks pretty much like a bug. It should be picking up on that constant without an issue. Feel free to work on it! 
:) I'm these weeks pretty much locked up in work, so can't work on it myself quickly.

**Comment by akosglue** [comment]
It might be solved in the meantime. I have set up the same module configuration as in your issue @iranimij 
One module in the app/code folder, like a traditional module. 
The other one is in the src folder, loaded by psr-4 via the magento composer.json (the modules, of course, have their own composer.json as well)
Then I added a static constant reference in the module in the app/code folder. It is working obviously because I tested it. 
Then I ran the dependency checker and it listed the module that sits in the src folder.
Bottom line: I cannot reproduce the issue, it works as intended.
